File: jaepyo99/[PGS]점 찍기.java
import java.util.*;
class Solution {
    public long solution(int k, int d) {
        long answer = 0;
        long limit=(long)d*d;
        for(int i=0;i<=d;i+=k){
            long ii=(long)i*i;
            long point=limit-ii;
            int top=(int)Math.sqrt(point);
            answer+= (top/k)+1;
        }
        return answer;
    }
}

File: hall_of_fame/gusah009/Kotlin/[PGS]점찍기.kt
import kotlin.math.sqrt

class Solution {
    fun solution(k: Int, d: Int): Long {
        var answer = 0L
        for (i in 0..d step k) {
            val height = sqrt(((d + i).toLong() * (d - i).toLong()).toDouble()).toInt()
            answer += height / k + 1
        }
        return answer
    }
}